cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -13,6 +20,25 @@ class StdStringCStr extends TaintFunction {
   }
 }
 
+/**
+ * The `std::string` function `operator+`.
+ */
+class StdStringPlus extends TaintFunction {
+  StdStringPlus() {
+    this.hasQualifiedName("std", "operator+") and
+    this.getParameter(0).getType().getUnspecifiedType().(ReferenceType).getBaseType() = any(StdBasicString s).getAnInstantiation()
```

Comment:
There are some overrides where the `std::basic_string` is on the right hand side with a single character or a C-style string on the left hand side.

Reply:
What they all have in common is that they _return_ a `std::basic_string`, so maybe we can use that. How about replacing `     this.getParameter(0).getType().getUnspecifiedType().(ReferenceType).getBaseType()` with `this.getType()`?